specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2022-11-30-preview/definitions/MigrateMySqlAzureDbForMySqlOfflineTask.json

```diff
@@ -0,0 +1,500 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Database Migration Service Resource Provider",
+    "version": "2022-11-30-preview"
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "MigrateMySqlAzureDbForMySqlOfflineTaskInput": {
+      "type": "object",
+      "description": "Input for the task that migrates MySQL databases to Azure Database for MySQL for offline migrations",
+      "properties": {
+        "sourceConnectionInfo": {
+          "description": "Connection information for source MySQL",
+          "$ref": "./Common.json#/definitions/MySqlConnectionInfo"
+        },
+        "targetConnectionInfo": {
+          "description": "Connection information for target Azure Database for MySQL",
+          "$ref": "./Common.json#/definitions/MySqlConnectionInfo"
+        },
+        "selectedDatabases": {
+          "type": "array",
+          "description": "Databases to migrate",
+          "items": {
+            "$ref": "#/definitions/MigrateMySqlAzureDbForMySqlOfflineDatabaseInput"
+          }
+        },
+        "makeSourceServerReadOnly": {
+          "type": "boolean",
+          "description": "Setting to set the source server read only",
+          "default": false
+        },
+        "startedOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Parameter to specify when the migration started"
+        },
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Optional resource Id of the source server if it is an azure instance"
+        },
+        "targetServerResourceId": {
+          "type": "string",
+          "description": "Optional resource Id of the target server"
+        },
+        "optionalAgentSettings": {
+          "type": "object",
+          "description": "Optional parameters for fine tuning the data migration",
+          "properties": {
+            "enableConsistentBackup": {
+              "type": "boolean",
+              "description": "Setting to enable transactional consistency",
+              "default": false
+            }
+          },
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "migrateAllViews": {
+          "type": "boolean",
+          "description": "If true, all view definitions will be migrated in the selected databases",
```

Comment:
>If true, all view definitions will be migrated in the selected databases [](http://example.com/codeflow?start=26&length=72)

What happens if false?

Reply:
@mentat9, I will update this description and the similar ones, but I am afraid I can't implement the other suggestions although I like most of them. The problem is that this PR documents the existing API which is already used by our web UI at the Azure portal, and the Azure CLI as well. I have bumped up the version number in this PR, but this is how the currently deployed version works. This API is not really meant to be used by the external customers except the rare cases when we instruct them to use scripting tools like Powershell to start the migration, and we do not use any clients generated from this spec internally. For the past few years, the developers modified it as they saw fit, and this is what we have right now.